js/ui/map.js

```diff
@@ -287,6 +287,50 @@ util.extend(Map.prototype, /** @lends Map.prototype */{
     },
 
     /**
+     * Set the map's minimum zoom level, and zooms map to that level if it is currently below it. If no parameter provided, unsets the current minimum zoom (sets it to 0)
+     * @param {zoom} any number between 0 and 20
+     * @returns {Map} `this`
+     */
+    setMinZoom: function(minZoom) {
+
+        var zoom = minZoom === undefined ? 0 : minZoom;
```

Comment:
To unset the minZoom, we want users to write

```js
map.setMinZoom(null) // clear and self-documenting
```

We do NOT want users to write

```js
map.setMinZoom() // wat?
```

I don't think this is anywhere near as clear a distinction as you lay out, way less than wat vs self-documenting. What about -Infinity? Or NaN? undefined vs null is not night and day.

If you want something clear, `map.removeMinZoom()`